src/nifgen/system_tests/test_system_nifgen.py

```diff
@@ -444,6 +322,65 @@ def test_send_software_edge_trigger_script(self, session):
         with session.initiate():
             session.send_software_edge_trigger(nifgen.Trigger.SCRIPT, 'ScriptTrigger0')
 
+    def test_get_channel_name(self, session):
+        name = session.get_channel_name(1)
+        assert name == '0'
+
+    def test_create_waveform_wrong_type(self, session):
+        for data in invalid_waveforms:
+            try:
+                session.create_waveform(data)
+                assert False
+            except (TypeError, ValueError):
+                pass
+
+
+class TestLibrary(SystemTests):
+    @pytest.fixture(scope='function')
+    def session(self):
+        with nifgen.Session('', '0', False, 'Simulate=1, DriverSetup=Model:5433 (2CH);BoardType:PXIe') as simulated_session:
+            yield simulated_session
+
+    @pytest.fixture(scope='function')
+    def session_5421(self):
+        with daqmx_sim_db_lock:
+            simulated_session = nifgen.Session('', '0', False, 'Simulate=1, DriverSetup=Model:5421;BoardType:PXI')
+        yield simulated_session
+        with daqmx_sim_db_lock:
+            simulated_session.close()
+
+    def test_error_message(self):
+        try:
+            # We pass in an invalid model name to force going to error_message
+            with nifgen.Session('', '0', False, 'Simulate=1, DriverSetup=Model:invalid_model (2CH);BoardType:PXIe'):
+                assert False
+        except nifgen.Error as e:
+            assert e.code == -1074134944
+            assert e.description.find('Insufficient location information or resource not present in the system.') != -1
+
+    # Test doesn't run over gRPC due to incorrect attribute name for attribute_value.
+    def test_channels_rep_cap(self):
+        with nifgen.Session('', '', False, 'Simulate=1, DriverSetup=Model:5433 (2CH);BoardType:PXIe') as session:
+            session.func_amplitude = 0.5
+            assert session.channels[0:1].func_amplitude == 0.5
+
+            session.channels[0].func_amplitude = 1
+            assert session.channels[0].func_amplitude == 1
+            assert session.channels[1].func_amplitude == 0.5
+
+    ''' Removed due to OSP disabled - #891
+    def test_fir_filter_coefficients(self):
+        with nifgen.Session('', '0', False, 'Simulate=1, DriverSetup=Model:5441;BoardType:PXI') as session:
+            coeff_array = [0 for i in range(95)]
+            coeff_array[0] = -1.0
+            coeff_array[2] = 1.0
+            session.configure_custom_fir_filter_coefficients(coeff_array)
+            session.commit()
+            array = session.get_fir_filter_coefficients()
+            assert len(array) == len(coeff_array)
+            assert array == coeff_array
+    '''
```

Comment:
Why did you move this?
Is there some reason you don't expect it to work even if #891 is resolved?

Reply:
If so, I think you should add a comment like you did for the other tests.

Reply:
Nevermind, I see you put a modified copy of it in the other class, too.